ambry-frontend/src/main/java/com/github/ambry/frontend/AmbrySecurityService.java

```diff
@@ -160,6 +160,18 @@ public void postProcessRequest(RestRequest restRequest, Callback<Void> callback)
     }
   }
 
+  @Override
+  public void processRequestCharges(RestRequest restRequest, RestResponseChannel responseChannel, BlobInfo blobInfo) {
+    Map<QuotaName, Double> requestCost = requestCostPolicy.calculateRequestCost(restRequest, responseChannel, blobInfo)
+        .entrySet()
+        .stream()
+        .collect(Collectors.toMap(entry -> QuotaName.valueOf(entry.getKey()), entry -> entry.getValue()));
+    setRequestCostHeader(requestCost, responseChannel);
+    if (quotaManager != null) {
+      quotaManager.charge(restRequest, blobInfo, requestCost);
+    }
+  }
```

Comment:
If this logic will need to be copied verbatim in closed impl, I'd rather see it in a method that can be called.

Reply:
Per discussion, this will be refactored in a follow up PR with charge logic moved to RestRequestService.